<launch>
  <arg name="dataset_name"   value="EUROC"/>
  <arg name="cache_time"     default="144.0"/>
  <arg name="path_to_vocab"
       default="/home/marcus/catkin_ws/src/Kimera-VIO/vocabulary/ORBvoc.yml"/>

  <arg name="body_frame_id"   value="base_link"/>
  <arg name="world_frame_id"  value="world"/>
  <arg name="map_frame_id"    value="map"/>
  <arg name="log_output"      default="false" />
  <arg name="visualize"       default="false"/>
  <arg name="error_to_scale"  default="false"/>

  <arg name="gnd_truth_topic"       value="/vicon/firefly_sbx/firefly_sbx"/>
  <arg name="left_cam_topic"        value="/cam0/image_raw"/>
  <arg name="right_cam_topic"       value="/cam1/image_raw"/>
  <arg name="lcd_input_topic"       value="lcd_input"/>
  <arg name="closure_result_topic"  value="result"/>
  <arg name="closure_image_topic"   value="image"/>
  <arg name="error_topic"           value="error"/>
  <arg name="markers_topic"         value="marker_array"/>
  <arg name="poses_topic"           value="pose_array"/>

  <include file="$(find kimera_lcd_ros)/launch/lcd_ros.launch"
    pass_all_args="true"/>

  <!-- Launch Transfrom Converter Node -->
  <node ns="lcd" name="transform_converter_node" pkg="kimera_lcd_ros"
        type="transform_converter_node.py" output="screen">

    <param name="gnd_truth_topic" value="$(arg gnd_truth_topic)"/>
    <param name="body_frame_id" value="$(arg body_frame_id)"/>
    <param name="world_frame_id" value="$(arg world_frame_id)"/>
  </node>
</launch>
